### PR TITLE
Attachment extension

### PR DESCRIPTION
**Pull request by MissAllSunday**
Add a generic binary extension to files saved in attachments dir

### CODE REVIEW

**Comment by Arantor** [comment]
Do we need to do anything with the upgrader if we merge this in?

File: Sources/Subs.php [hunk 1]
```diff
@@ -3395,7 +3401,21 @@ function getAttachmentFilename($filename, $attachment_id, $dir = null, $new = fa
...
-	return $path . '/' . $attachment_id . '_' . $file_hash;
+	// Add a generic extension to the full path
+	$return_file = $path . '/' . $attachment_id . '_' . $file_hash . $FileZilla_ext;
+
+	// Is this an old attachment?
+	if (!file_exists($return_file))
+	{
+		// Lets check if there is one without the extension
+		if(file_exists($path . '/' . $attachment_id . '_' . $file_hash))
+		{
```
Comment (live627) [inline Sources/Subs.php:3412]: I think these two if() statements can be combined.

**Comment by jdarwood007** [comment]
What is the purpose of this?  We drop extensions for a reason, so why are we adding one back, even a generic version?

**Comment by Arantor** [comment]
We drop the original extension to prevent the file being handled by the system in case the attachments folder is 1) web accessible and 2) normal .htaccess isn't going to work for whatever reason (file missing, not using Apache, whatever)... this means files that are .php can't be executed as PHP which would be a bad security issue.

On the other hand, we have _extremely regular_ issues with support because of FileZilla. FileZilla, by default, treats any file without an extension as a text file. Regardless of content. That means any time a default installation of FileZilla touches the SMF attachments folder, the backup will be irretrievably damaged unless _everything_ is a text file (and invariably isn't)

The best way to solve both problems is to have a generic extension for all files, that won't be executed as PHP or similar and will at the same time prevent FileZilla doing what FileZilla does (which has been filed as a bug with FileZilla for _years_ but they assert that it is not a bug)

This is not even remotely a new problem, I draw your attention to http://www.simplemachines.org/community/index.php?topic=374178.0

**Comment by NanoSector** [comment]
So what's wrong with this commit now?

**Comment by MissAllSunday** [comment]
Still needs some testing, the code seems simple too, perhaps another check or another approach would fit best.

We also need to decide what to do on upgrading.

File: Sources/Subs.php (hunk 1 above)
Comment (MissAllSunday) [inline Sources/Subs.php:3412]: Yep! :) let me see if I still have the branch on local.

See "help squad"? I didn't overreact! :P

**Comment by Arantor** [comment]
The upgrader should fix all files. It should also back fill the mime type and file hash columns.

**Pull request closed without merging**